webharness/harness.js

```diff
@@ -60,6 +63,62 @@ function socket_message(data) {
 }
 
 /*
+ * Receive a single message from |socket|. If
+ * there is a deferred (from wait_for_message)
+ * waiting on it, resolve that deferred. Otherwise
+ * queue the message for a future waiter.
+ */
+function socket_message_test(data) {
```

Comment:
I think this function should be re-factored so that it shares one common function with the socket_message() function above, to avoid code duplication. socket_message() and socket_message_test() would then become thin wrappers around the shared function which only pass the appropriated socket into the shared function.